xds/src/main/java/io/grpc/xds/ClientXdsClient.java

```diff
@@ -131,6 +140,7 @@
   private static final String TYPE_URL_FILTER_CONFIG =
       "type.googleapis.com/envoy.config.route.v3.FilterConfig";
 
+  private final FilterRegistry filterRegistry = FilterRegistry.getDefaultRegistry();
```

Comment:
nit: why add this `filterRegistry` variable ? It can't be for testing because of this inline assignment and I don't see any gain (other than syntactic sugar of using `filterRegistry` instead of `FilterRegistry.getDefaultRegistry()`).

Reply:
Hmm... We put real filter implementations into the FilterRegistry used for test. So behavior-wise, there is no difference. Making the FilterRegistry an argument of static method `parseHttpFilter()` improves the test code's readability, robustness and extensibility. It allows the test code to control the registry used for testing, so the readers can easily know where the filter implementations come from. Overall, this code style better modularize static methods by breaking the dependency of the global singleton registry.

Reply:
> Overall, this code style better modularize static methods by breaking the dependency of the global singleton registry.

`processClientSideListener` changed from static to non-static and it now references the member variable `filterRegistry`. As a static method and a parameter for `filterRegistry` the testability is better. But this is a minor point.